kauri/kauri_test.go

```diff
@@ -0,0 +1,64 @@
+package kauri
+
+import (
+	"testing"
+
+	"github.com/relab/hotstuff"
+	"github.com/relab/hotstuff/crypto"
+	"github.com/relab/hotstuff/crypto/ecdsa"
+)
+
+func TestSubTree(t *testing.T) {
+	subTreeTestData := []struct {
+		test_a   []hotstuff.ID
+		test_b   []hotstuff.ID
+		expected bool
+	}{
+		{[]hotstuff.ID{1, 2, 3, 4, 5}, []hotstuff.ID{1, 2, 3, 4, 5}, true},
+		{[]hotstuff.ID{1, 2, 3, 4, 5}, []hotstuff.ID{1, 2, 3, 4}, false},
+		{[]hotstuff.ID{1, 2, 3, 4, 5}, []hotstuff.ID{1, 2, 3, 4, 6}, false},
+		{[]hotstuff.ID{1, 2, 3, 4, 5}, []hotstuff.ID{1, 2, 3, 4, 5, 6}, true},
+		{nil, nil, true},
+	}
+	for _, test := range subTreeTestData {
+		if isSubSet(test.test_a, test.test_b) != test.expected {
+			t.Errorf("SubTree(%v, %v) = %v; want %v", test.test_a, test.test_b, !test.expected, test.expected)
+		}
+	}
+}
+
+func TestCanMerge(t *testing.T) {
+	mergeTestData := []struct {
+		test_a   []hotstuff.ID
+		test_b   []hotstuff.ID
+		expected bool
```

Comment:
Same for `a` and `b` as comment above. Replace with `wantErr` to more clearly signify that the bool represents that we want an error or not. See other examples; we should try to follow the same pattern.